### PR TITLE
Single root logger handler :)

### PR DESCRIPTION
**Pull request by sneakers-the-rat**
Fix: https://github.com/Aharoni-Lab/miniscope-io/issues/69

Instead of having each logger having their own handler, we instead have one single root logger `miniscope_io` that has all the handlers, and then all the child loggers propagate messages up to it.

The biggest change in behavior here is that we now will have a single log file (or, rather, `n` log files depending on how many rotating files we have configured) rather than a log file per logger. If we prefer that, we can go back to that, but it requires us to also have multiple stdout handlers since we need to set `propagate = false` at the logger level to prevent double-logging, which we have mostly avoided by chance from not having any nested loggers.

Opening this as a draft for the moment because I think this will play screwy with multiprocessing, so let me test that real quick too.

edit: ok added multiprocessing handling - if a logger is spawned in a child process, it logs to a separate file that includes the logger's name as well as the PID (anticipating people wanting to run multiple of the same kind of miniscope on the same machine).

----
📚 Documentation preview 📚: https://miniscope-io--73.org.readthedocs.build/en/73/

### CODE REVIEW

File: miniscope_io/logging.py [hunk 1]
```diff
@@ -48,27 +49,86 @@ def init_logger(
...
         log_file_n = config.logs.file_n
     if log_file_size is None:
         log_file_size = config.logs.file_size
 
+    # set our logger to the minimum of the levels so that it always handles at least that severity
+    # even if one or the other handlers might not.
+    min_level = min([getattr(logging, level), getattr(logging, file_level)])
+
     if not name.startswith("miniscope_io"):
         name = "miniscope_io." + name
```
Comment (t-sasatani) [inline miniscope_io/logging.py:65]: Not really related here, but would it hurt to change this to `mio`? `miniscope_io` is pretty long for my small terminal

File: miniscope_io/logging.py [hunk 2]
```diff
@@ -48,27 +49,86 @@ def init_logger(
...
+            _file_handler(
+                name=f"{name}_{mp.current_process().pid}",
+                file_level=file_level,
+                log_dir=log_dir,
+                log_file_n=log_file_n,
+                log_file_size=log_file_size,
+            )
+        )
+        logger.addHandler(_rich_handler(level))
+        logger.propagate = False
```
Comment (t-sasatani) [inline miniscope_io/logging.py:90]: Oh, didn't know how to do this. Neat.

**Review by t-sasatani — APPROVED** [review]
I think it looks great and we can merge.

File: miniscope_io/logging.py (hunk 1 above)
Comment (sneakers-the-rat) [reply miniscope_io/logging.py:65]: yep! good idea. will do rn

Comment (sneakers-the-rat) [reply miniscope_io/logging.py]: done in: https://github.com/Aharoni-Lab/miniscope-io/pull/73/commits/3624fd8aa9c6cdc98d8058bc9b7053d1c32d7628

**Pull request merged**